studio/src/editors/merch-card-editor.js

```diff
@@ -50,76 +62,81 @@ class MerchCardEditor extends LitElement {
         );
     }
 
-    connectedCallback() {
-        super.connectedCallback();
-        this.addEventListener('keydown', this.#handleKeyDown);
-    }
-
-    disconnectedCallback() {
-        super.disconnectedCallback();
-        this.removeEventListener('keydown', this.#handleKeyDown);
-    }
+    render() {
+        if (this.loading)
+            return html`<div id="editor">
+                <sp-progress-circle indeterminate size="l"></sp-progress-circle>
+            </div>`;
 
-    #handleKeyDown(e) {
-        if (e.key === 'Escape') {
-            e.stopPropagation();
-        }
-    }
+        if (this.refreshing || !this.fragment) return nothing;
 
-    render() {
         if (this.fragment.model.path !== MODEL_PATH) return nothing;
+
         const form = Object.fromEntries([
             ...this.fragment.fields.map((f) => [f.name, f]),
         ]);
-        return html`
-            <p>${this.fragment.path}</p>
+
+        return html`<div id="editor">
+            <editor-panel
```

Comment:
this should be the other way around, editor-panel contains merch-card-editor.
also,`editor-panel` and  fragment-editor-base.js should be combined together.

e.g: https://github.com/adobecom/mas/blob/main/studio/src/editor-panel.js#L279